main.cpp

```diff
@@ -117,17 +132,25 @@ struct <#structname2#>
 
 int main()
 {
-    T <#name1#>( , );                                             //6
-    T <#name2#>( , );                                             //6
+    T first(3.4, "name1");                                             //6
+    T second(4.5, "name2" );                                             //6
     
-    <#structName1#> f;                                            //7
-    auto* smaller = f.compare( , );                              //8
-    std::cout << "the smaller one is << " << smaller->name << std::endl; //9
+    Comparison f;                                            //7
+    auto* smaller = f.compare(&first, &second);    //8
+
+    if (smaller != nullptr)
+    {
+        std::cout << "the smaller one is << " << smaller->name << std::endl; //9
+    }
+    else
+    {
+        std::cout << "the values are the same" << std::endl;
```

Comment:
There are multiple reasons that the compare function can return a `nullptr`
- [ ] revise